initramfs/init.sh

```diff
@@ -241,19 +270,32 @@ check_and_set_article_number()
             return 0
         fi
 
-        if [ ! -r "${article_number_file}" ]; then
+        if [ ! -r "${article_number_file}" ] && [ ! -r "${country_code_lock_file}" ]; then
             umount "${dev}"
-            echo "No article number file found on '${dev}', skipping."
+            echo "No article number file or country code lock file found on '${dev}', skipping."
             return 0
         fi
 
-        article_number="$(cat "${article_number_file}")"
-        echo "Trying to write article nr: '${article_number}'."
-        # shellcheck disable=SC2086
-        if ! i2ctransfer -y 3 w6@0x57 0x01 0x00 ${article_number}; then
-            umount "${dev}"
-            echo "Failed to write article number to EEPROM, skipping."
-            return 0
+        if [ -r "${article_number_file}" ]; then
+            article_number="$(cat "${article_number_file}")"
+            echo "Trying to write article nr: '${article_number}'."
+            # shellcheck disable=SC2086
+            if ! i2ctransfer -y 3 w6@0x57 0x01 0x00 ${article_number}; then
+                umount "${dev}"
```

Comment:
You cannot unmount the usb stick at this point, otherwise you cannot read the country code below.
I would do the first 2 checks and, if file found, do your thing, otherwise output a message saying file not found and, in the end, umount the usb stick.

I added the branch `personal/alex/SEM-146-country_code_lock` with this change, but it looks like this:

```
        echo "Attempting to mount '${dev}'."
        if ! mount -t f2fs,ext4,vfat,auto -o exec,noatime "${dev}" "${ARTICLENR_USB_MOUNT}"; then
            return 0
        fi

        if [ -r "${article_number_file}" ]; then
            article_number="$(cat "${article_number_file}")"
            echo "Trying to write article nr: '${article_number}'."
            # shellcheck disable=SC2086
            if ! i2ctransfer -y 3 w6@0x57 0x01 0x00 ${article_number}; then
                echo "Failed to write article number to EEPROM, skipping."
            fi
        else
            echo "No article number file found on '${dev}', skipping."
        fi

        if [ -r "${country_code_lock_file}" ]; then
            country_code="$(cat "${country_code_lock_file}")"
            echo "Trying to write country code lock: '${country_code}'."
            # shellcheck disable=SC2086
            if ! i2ctransfer -y 3 w4@0x57 0x01 0x18 ${country_code}; then
                echo "Failed to write country code lock to EEPROM, skipping."
            fi
        else
            echo "No country code lock file found on '${dev}', skipping."
        fi

        umount "${dev}"
        retries=0
```

Reply:
Thanks again @alexborro . Incorporated in e160b4c24239d97d5adeefd03caac937005d624e